drivers/watchdog/wdt_nrfx_wdt.c

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (c) 2018, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <nrfx_wdt.h>
+#include <nrf_clock.h>
+#include <clock_control.h>
+#include <watchdog.h>
+
+#define SYS_LOG_DOMAIN "WDT NRF"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_WDT_LEVEL
+#include <logging/sys_log.h>
+
+#define NRF_WDT_CHANNELS_NUMBER 8
+
+static struct device DEVICE_NAME_GET(wdt_nrf);
+
+/* There are eight reload registers (channels) in Nordic's WDT. */
+static nrfx_wdt_channel_id m_ch[NRF_WDT_CHANNELS_NUMBER];
```

Comment:
Now that `nrfx_wdt_channel_feed` is called with `(nrfx_wdt_channel_id)channel_id` as the parameter, we should get rid of this to not create a mess.